### PR TITLE
Update database : Duels procedures and adjustments

### PR DESCRIPTION
**Pull request by FrancoisPog**
## Changes

- [x] Procedures : `create` / `get` / `getAll` duels
- [x] Add a boolean `inProgress` to duels
- [x] Set users answers  as `JSON`
- [x] Tests

### CODE REVIEW

File: server/db/database.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ dotenv.config();
 const Database = {};
 
 // Be sure to add a new version at the end of this array (it must be sorted)
-const versions = ["2021-01-08", "2021-01-14", "2021-01-16", "2021-01-21"];
+const versions = ["2021-01-08", "2021-01-14", "2021-01-16", "2021-01-21", "2021-01-24"];
```
Comment (nathanaelhoun) [inline server/db/database.js:13]: We're on the 25th :stuck_out_tongue:

**Review by valentinperignon — APPROVED** [review]
LGTM!

File: server/test/database.test.js [hunk 2]
```diff
@@ -184,3 +186,69 @@ describe("Procedures", () => {
...
+
+describe("Procedure duels", () => {
+  const duelIds = [];
+
+  before("Clear duels, results & users", (done) => {
+    forceTruncateTables("results", "duel", "user").then(() => insertData("users.sql").then(done));
+  });
+
+  it("Create a duel", async () => {
+    let res = await queryPromise("CALL createDuel(?,?,?);", ["fpoguet", "nhoun", JSON.stringify(["questions"])]);
```
Comment (nathanaelhoun) [inline server/test/database.test.js:198]: I will crush you!

File: server/test/database.test.js [hunk 3]
```diff
@@ -184,3 +186,69 @@ describe("Procedures", () => {
...
+
+  it("Get a noexistent duel", async () => {
+    const res = await queryPromise("CALL getDuel(?,?);", [-1, "fpoguet"]);
+    expect(res[0]).to.have.length(0);
+  });
+
+  it("Get a duel of others players", async () => {
+    const res = await queryPromise("CALL getDuel(?,?);", [duelIds[0], "vperigno"]);
+    expect(res[0]).to.have.length(0);
+  });
```
Comment (nathanaelhoun) [inline server/test/database.test.js:231]: I don't understand the purpose of this test

**Review by nathanaelhoun — APPROVED** [review]
Just one small question

File: server/db/database.js (hunk 1 above)
Comment (FrancoisPog) [reply server/db/database.js:13]: oupsi !

File: server/test/database.test.js (hunk 2 above)
Comment (FrancoisPog) [reply server/test/database.test.js:198]: 🖖

File: server/test/database.test.js (hunk 3 above)
Comment (FrancoisPog) [reply server/test/database.test.js:231]: It checks that if a user requests an existing duel but doesn't participate, they should not return anything.

Comment (nathanaelhoun) [reply server/test/database.test.js:231]: Oh ok! That's nice :-)

**Pull request merged**